lib/spack/spack/main.py

```diff
@@ -870,6 +871,41 @@ def restore_macos_dyld_vars():
             os.environ[dyld_var] = os.environ[stored_var_name]
 
 
+def resolve_alias(cmd_name, cmd):
+    """Resolves aliases in the given command.
+
+    Args:
+        cmd_name (string): command name.
+        cmd (list): command line arguments.
+
+    Returns:
+        (str, list): new command name and arguments.
+    """
+    if cmd_name not in spack.cmd.all_commands():
+        aliases = spack.config.get("config:aliases")
+        alias = None
+
+        if aliases:
+            alias = aliases.get(cmd_name)
+
+        if alias is not None:
+            alias_parts = shlex.split(alias)
+            cmd_name = alias_parts[0]
+            cmd = alias_parts + cmd[1:]
+        else:
+            # If cmd_name is not an alias but also not a proper command,
+            # check whether it is a prefix of a proper command.
+            prefix = [c for c in spack.cmd.all_commands() if c.startswith(cmd_name)]
+
+            # We will use the found command only if there is exactly one
+            # match.
+            if len(prefix) == 1:
+                cmd_name = prefix[0]
+                cmd = [cmd_name] + cmd[1:]
```

Comment:
I'd be ok with a better "did you mean?" error message for this, but I don't think we should automatically run a command the user didn't actually type, even if the prefix is unique.  We have completion for that.

Rationale: I think this can lead to sloppy scripting. It allows people to write, e.g., `spack inf hdf5`in a script and not get an error, which may one day become ambiguous. We should fail fast if you're not using valid commands.

Reply:
I have removed that part for now. I can give a separate "did you mean" functionality a try in a separate PR.

Reply:
Yeah let’s save “did you mean” for a separate PR, as I think people will have opinions.